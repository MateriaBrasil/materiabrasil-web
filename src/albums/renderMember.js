import React from 'react'
import Dialog from 'materials/Dialog'
import { Formik } from 'formik'
import FormControl from '@material-ui/core/FormControl'
import Button from '@material-ui/core/Button'
import Input from '@material-ui/core/Input'
import InputLabel from '@material-ui/core/InputLabel'
import FormHelperText from '@material-ui/core/FormHelperText'
import Redirect from 'react-router-dom/Redirect'
import { New } from 'croods'
import Error from 'Error'

export default props => routeProps => {
  const validate = values => {
    let errors = {}

    if (!values.email) {
      errors.email = 'Required'
    }

    return errors
  }

  return (
    <Dialog {...props} {...routeProps} title="Adicione um membro ao álbum">
      {props.currentUser && (
        <New
          name="albumUsers"
          path="/album_users"
          render={({ create, creating, error }) => {
            return (
              <Formik
                initialValues={{ email: '' }}
                validate={validate}
                onSubmit={values => {
                  create({
                    user_email: values.email,
                    album_id: parseInt(routeProps.match.params.id, 10),
                  })
                }}
                render={({
                  values,
                  errors,
                  touched,
                  handleChange,
                  handleBlur,
                  handleSubmit,
                }) => (
                  <form onSubmit={handleSubmit}>
                    <FormControl
                      error={touched['email'] && !!errors['email']}
                      aria-describedby="email-error-text"
                      fullWidth
                    >
                      <InputLabel htmlFor="email-error">
                        Email do membro
                      </InputLabel>
                      <Input
                        id="email"
                        name="email"
                        type="email"
                        onChange={handleChange}
                        onBlur={handleBlur}
                        value={values.email}
                      />
                      <FormHelperText id="email-error-text">
                        {touched['email'] && errors['email']}
                      </FormHelperText>
                    </FormControl>
                    {error && <Error>{error}</Error>}
                    <Button type="submit">Adicionar</Button>
                  </form>
                )}
              />
            )
          }}
          renderCreated={() => (
            <Redirect
              to={{
                pathname: `/albums/${routeProps.match.params.id}/success`,
                state: {
                  title: 'Usuário adicionado',
                  message: 'Parabéns! Você adicionou o usuário com sucesso.',
                },
              }}
            />
          )}
        />
      )}
    </Dialog>
  )
}
