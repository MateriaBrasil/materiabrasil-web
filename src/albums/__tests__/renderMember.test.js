import React from 'react'
import renderer from 'react-test-renderer'

import renderMember from '../renderMember'

jest.mock('materials/Dialog', () => props => (
  <div {...props}>Dialog - {props.children}</div>
))

jest.mock('croods', () => ({
  New: props => (
    <div {...props}>
      New - {props.render({ create: () => {} })} - {props.renderCreated()}
    </div>
  ),
}))

jest.mock('formik', () => ({
  Formik: props => {
    props.validate({ email: 'fesg' })
    props.validate({ email: '' })
    props.onSubmit({ email: 'fesg' })

    return (
      <div {...props}>
        New -{' '}
        {props.render({ values: { email: '123' }, touched: {}, errors: {} })}
      </div>
    )
  },
}))

jest.mock('../../payments/form/render', () =>
  jest.fn(props => routeProps => <div {...props}>render</div>),
)

jest.mock('react-router-dom/Redirect', () => props => (
  <div {...props}>Redirect</div>
))

const routeProps = { match: { params: { id: 123 } } }
const props = {
  currentUser: { albums: [{ id: 1, name: '123' }, { id: 2, name: '345' }] },
}

describe('render', () => {
  it('renders correctly', () => {
    const tree = renderer.create(renderMember(props)(routeProps)).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
