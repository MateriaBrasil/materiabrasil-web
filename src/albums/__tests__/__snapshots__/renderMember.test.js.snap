// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<div
  currentUser={
    Object {
      "albums": Array [
        Object {
          "id": 1,
          "name": "123",
        },
        Object {
          "id": 2,
          "name": "345",
        },
      ],
    }
  }
  match={
    Object {
      "params": Object {
        "id": 123,
      },
    }
  }
  title="Adicione um membro ao Ã¡lbum"
>
  Dialog - 
  <div
    name="albumUsers"
    path="/album_users"
    render={[Function]}
    renderCreated={[Function]}
  >
    New - 
    <div
      initialValues={
        Object {
          "email": "",
        }
      }
      onSubmit={[Function]}
      render={[Function]}
      validate={[Function]}
    >
      New
    </div>
     - 
    <div
      to="/albums/123/member/success"
    >
      Redirect
    </div>
  </div>
</div>
`;
